packages/studio-base/src/panels/ThreeDimensionalViz/TopicTree/VisibilityToggle.tsx

```diff
@@ -1,68 +1,48 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
-//
-// This file incorporates work covered by the following copyright and
-// permission notice:
-//
-//   Copyright 2018-2021 Cruise LLC
-//
-//   This source code is licensed under the Apache License, Version 2.0,
-//   found at http://www.apache.org/licenses/LICENSE-2.0
-//   You may not use this file except in compliance with the License.
 
-import { ITheme, useTheme } from "@fluentui/react";
-import BlockHelperIcon from "@mdi/svg/svg/block-helper.svg";
+import BlockIcon from "@mui/icons-material/Block";
+import { IconButton, SvgIcon, Tooltip, Theme } from "@mui/material";
+import { createStyles, makeStyles } from "@mui/styles";
+import cx from "classnames";
 import { useCallback } from "react";
-import styled from "styled-components";
-import tinyColor from "tinycolor2";
 
 import { Color } from "@foxglove/regl-worldview";
-import Icon from "@foxglove/studio-base/components/Icon";
 import { defaultedRGBStringFromColorObj } from "@foxglove/studio-base/util/colorUtils";
-import { colors } from "@foxglove/studio-base/util/sharedStyleConstants";
 
-import { ROW_HEIGHT } from "./constants";
-
-export const TOPIC_ROW_PADDING = 3;
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    unavailable: {
+      cursor: "not-allowed",
+    },
+    button: ({ overrideRGB, size }: StyleProps) => ({
+      fontSize: size === "small" ? 10 : 12,
+      color: `${overrideRGB ?? theme.palette.action.active} !important`,
+      position: "relative",
 
-export const TOGGLE_WRAPPER_SIZE = 24;
+      "&:hover, &:focus": {
+        backgroundColor: theme.palette.action.hover,
 
-export const TOGGLE_SIZE_CONFIG = {
-  NORMAL: { name: "NORMAL", size: 12 },
-  SMALL: { name: "SMALL", size: 10 },
-};
-
-const SToggle = styled.label`
-  width: ${TOGGLE_WRAPPER_SIZE}px;
-  height: ${TOGGLE_WRAPPER_SIZE}px;
-  display: inline-flex;
-  align-items: center;
-  justify-content: center;
-  position: relative;
-  cursor: pointer;
-  outline: 0;
-  span {
-    border: 1px solid ${({ theme }) => theme.palette.neutralLight} !important;
-  }
-  :hover {
-    background-color: rgba(255, 255, 255, 0.1);
-  }
-  :focus {
-    span {
-      border: 1px solid ${colors.BLUE} !important;
-    }
-  }
-`;
-
-const SSpan = styled.span`
-  :hover {
-    transform: scale(1.2);
-  }
-`;
+        "& circle": {
+          stroke: overrideRGB ?? theme.palette.info.main,
+          strokeWidth: 4,
+        },
+      },
+      "&:active": {
+        color: theme.palette.info.main,
+      },
+    }),
+    circle: ({ checked, visibleInScene }: StyleProps) => ({
+      stroke: "currentColor",
+      strokeWidth: 2,
+      fill: checked ? "currentColor" : "none",
+      fillOpacity: visibleInScene ? 0.8 : theme.palette.action.disabledOpacity,
+    }),
+  }),
+);
 
-export type Size = keyof typeof TOGGLE_SIZE_CONFIG;
-type Props = {
+export type IVisibilityToggle = {
```

Comment:
```suggestion
export type VisibilityToggleProps = {
```

Its not an interface...